src/Storage/LakatStorageStub.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace MediaWiki\Extension\Lakat\Storage;
+
+use Exception;
+
+class LakatStorageStub implements LakatStorageInterface {
+	private static LakatStorageStub $instance;
+
+	public static function getInstance(): LakatStorageStub {
+		if ( !isset( self::$instance ) ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	public function createBranch( string $name, array $options ): string {
+		$branchId = 'branch_' . $this->getSlug( $name );
+
+		// create directory for this branch
+		$dir = $this->getBranchDir( $branchId );
+		if ( !@mkdir( $dir, 0755, true ) ) {
+			throw new Exception( 'Failed to create branch directory ' . $dir );
+		}
+
+		// store branch definition
+		if (
+			!file_put_contents(
+				$dir . '/.branch',
+				json_encode( compact( 'name' ) + $options, JSON_THROW_ON_ERROR )
+			)
```

Comment:
Is this the json-encoded branch-metadata, or just temporary storage?

Reply:
This is just a temporary stub for the remote storage. I had to simulate it somehow.